crane/models.py

```diff
@@ -100,18 +98,30 @@ def is_disconnected(self):
             return True
 
     def check_preconditions(self):
-        from . import settings  # avoiding circular imports
-
+        from . import hooks, settings  # avoiding circular imports
         try:
+            self.repo = git.Repo(environ['CI_PROJECT_DIR'])
             self.new_commit
-        except git.GitCommandError:
+        except gitdb.exc.BadName:
             click.secho(
                 f'The new version you specified, {self.new_version}, is not a valid git reference! '
-                'Please use a commit hash (or other ref) in your new_image tag.',
+                'crane is running on limited mode, all hooks have been disabled. '
+                'It is highly recommended you use Git references for your deployments.',
+                err=True,
+                fg='red',
+            )
+            hooks.deactivate()
```

Comment:
This feels like an antipattern, could you do it the other way around and set `self.is_limited = True` instead? Cause then:

- We avoid a module writing to another module's state, it'd be replaced with just one `deployment.is_limited` read in the `hooks` module, so that's one cross-module interaction instead of two, and it's a less scary read, rather than a write
- The `deployment` object is already used to distribute global state anyway

Reply:
yes, this makes sense to me